main.tf

```diff
@@ -113,7 +111,7 @@ resource "ibm_is_virtual_endpoint_gateway_ip" "endpoint_gateway_ip" {
     (gateway_ip.ip_name) => gateway_ip
   }
   gateway     = local.vpe_map[each.value.gateway_name].id
-  reserved_ip = ibm_is_subnet_reserved_ip.ip[each.key].reserved_ip
+  reserved_ip = lookup(var.reserved_ips, each.value.name, module.ip.reserved_ip_map[each.value.name])
```

Comment:
Need to double check some logic I think, now that `var.reserved_ips` is introduced:

The for_each loop for both `module.ip` and `resource.ibm_is_virturl_endpoint_gateway_ip` are both based on `local.endpoint_ip_list`.

If the consumer specifies existing reserved IPs created outside the module with `var.reserved_ips`, what is preventing the `module.ip` from also creating another reserved_ip for the same service?

You might need to limit `module.ip` to only generate new reserved entries for items in `local.endpiont_ip_list` that do NOT have an entry in `var.reserved_ips`? Or something similar?

Reply:
Would this [line](https://github.com/terraform-ibm-modules/terraform-ibm-vpe-gateway/pull/415/files#diff-9116576c0a8b977dbe986f7739366cfb48a4cfe0b1c3b7bf494929fd80965426R41) not limit `module.ip` to only generating reserved ips in `local.endpoint_ip_list` once there is no entry for that name already existing in `var.reserved_ips`.

Reply:
I think you are right, I think that should eliminate any duplicates.